mongodb_migrations/cli.py

```diff
@@ -19,105 +19,167 @@ class MigrationManager(object):
+                  (migration_label, file_to_apply))
+            try:
+                migration_object = self._get_migration_instance(file_to_apply)
+                if self.config.execution == Execution.UPGRADE:
                     migration_object.upgrade()
-                except Exception as e:
-                    print("Failed to upgrade version: %s" %
-                          self.migrations[migration_datetime])
-                    print(e.__class__)
-                    if hasattr(e, 'message'):
-                        print(e.message)
-                    sys.exit(1)
-                print("Succeed to upgrade version: %s" %
-                      self.migrations[migration_datetime])
-                self._create_migration(migration_datetime)
-
-    def _downgrade(self):
-        for migration_datetime in sorted(
-                self.database_migration_names, reverse=True):
-            if self.migrations[migration_datetime]:
-                print("Trying to downgrade version: %s" %
-                      self.migrations[migration_datetime])
-                try:
-                    module = __import__(self.migrations[migration_datetime])
-                    migration_object = module.Migration(
-                        host=self.config.mongo_host,
-                        port=self.config.mongo_port,
-                        database=self.config.mongo_database)
+                    self._create_migration(
+                        migration_label, self.config.execution)
+                else:
                     migration_object.downgrade()
-                except Exception as e:
-                    print("Failed to downgrade version: %s" %
-                          self.migrations[migration_datetime])
-                    print(e.__class__)
-                    if hasattr(e, 'message'):
-                        print(e.message)
-                    sys.exit(1)
-                print("Succeed to downgrade version: %s" %
-                      self.migrations[migration_datetime])
-                self._remove_migration(migration_datetime)
+                    self._create_migration(
+                        migration_label, self.config.execution)
+                    self._create_migration(
+                        self.migrations_to_apply[-1]
+                        if len(self.migrations_to_apply) == 2
+                        else None, self.config.execution)
+                    break
+            except Exception as e:
+                print("Failed to apply version: %s" % file_to_apply)
+                print(e.__class__)
+                if hasattr(e, 'message'):
+                    print(e.message)
+                sys.exit(1)
+            print("Succeed to apply version: %s" % file_to_apply)
 
     def _get_migration_names(self):
         return self.db.database_migrations.find().sort(
-            'migration_datetime', pymongo.DESCENDING)
+            'migration_label', pymongo.DESCENDING)
```

Comment:
This is a breaking change, there should be a way to provide an upgrade mechanism for those who use the framework and now their database got changed